Add warning about JEP 498 (#1691)
hazelcast/hz-docs: docs/modules/data-structures/pages/setting-data-format.adoc

Keep in mind that you should have already enabled HD memory usage for your cluster. See the xref:storage:high-density-memory.adoc#configuring-high-density-memory-store[Configuring High-Density Memory Store section].

Note that `NATIVE` memory stores data in binary format. Maps stored in the HD memory store have to be deserialized before they can be queried. A best practice is to use on-heap memory for maps that will be frequently queried when possible.

> NOTE: From JDK 24, Java gives a warning when using HD Memory due to https://openjdk.org/jeps/498[JEP 498]. To avoid this warning, add `--sun-misc-unsafe-memory-access=allow`  to your xref:configuration:jvm-parameters.adoc[Java command] when launching Hazelcast.

Reviewer: ```suggestion
NOTE: From JDK 24, Java gives a warning when using HD Memory due to https://openjdk.org/jeps/498[JEP 498]. To avoid this warning, add `--sun-misc-unsafe-memory-access=allow` to your xref:configuration:jvm-parameters.adoc[Java command] when launching Hazelcast.
```

Turbo-nit.

> NOTE Since JDK 24, Java will give a warning when using HD Memory due to https://openjdk.org/jeps/498[JEP 498]. To avoid those warnings, please add `--sun-misc-unsafe-memory-access=allow`  to your xref:../../configuration/jvm-parameters.adoc[Java command] when launching Hazelcast.

Reviewer 2: ```suggestion
NOTE: From JDK 24, Java gives a warning when using HD Memory due to https://openjdk.org/jeps/498[JEP 498]. To avoid this warning, add `--sun-misc-unsafe-memory-access=allow`  to your xref:configuration:jvm-parameters.adoc[Java command] when launching Hazelcast.
```
